airflow/models/dag.py

```diff
@@ -1466,6 +1466,8 @@ def fetch_callback(
             # context for the callback.
             if dag.partial:
                 tis = [ti for ti in tis if not ti.state == State.NONE]
+            # filter out removed tasks
+            tis = [ti for ti in tis if ti.state != TaskInstanceState.REMOVED]
```

Comment:
We should merge this with line 1468 above

Reply:
Done! 👍